src/subscription_manager/managercli.py

```diff
@@ -2660,6 +2662,43 @@ def _list(self, all_overrides, specific_repos):
             print(columnize(names, echo_columnize_callback, *values, indent=2) + "\n")
 
 
+class RoleCommand(CliCommand):
+    def __init__(self):
+        shortdesc = _("Modify system purpose role")
+        super(RoleCommand, self).__init__("role", shortdesc, primary=False)
+        self.parser.add_option("--set", dest="set_role",
+                               help=(_("Set role of system purpose")))
+        self.parser.add_option("--unset", dest="unset_role", action="store_true",
+                               help=(_("Unset role of system purpose")))
+
+    def _validate_options(self):
+        if self.options.set_role and self.options.unset_role:
+            system_exit(os.EX_USAGE, _("Error: Options --set and --unset of role subcommand are mutually exclusive."))
+
+    def _set_role(self, role):
+        ret = save_role_to_syspurpose_metadata(role)
+        if ret:
+            print(_("System Purpose role has been set to: %s") % role)
+        else:
+            print(_("Error: System Purpose role has NOT been set to: %s") % role)
+
+    def _unset_role(self):
+        return save_role_to_syspurpose_metadata(None)
```

Comment:
This should also use the boolean returned to print out something like 'System purpose role has been unset'.

Reply:
Good point. I will add printing of this information.